Userland/Libraries/LibCore/System.cpp

```diff
@@ -887,4 +887,12 @@ ErrorOr<Vector<gid_t>> getgroups()
     return groups;
 }
 
+ErrorOr<void> mkfifo(StringView pathname, mode_t mode)
+{
+    String path_string = pathname;
```

Comment:
If you re-order the commits, we should be able to avoid the allocation here like we do in most of the other System wrappers.

Looking at the LibC `mkfifo(..)` implementation it just a wraps `mknod(..)`. So if we move the `mknod(..)` wrapper commit before this one, then we can pass the `mkfifo(..)` arguments through to `mknod(..)` wrapper, keeping the `StringView` and avoiding the `String` conversion (and allocation) all the way through.